### ISSUE TITLE
Implicit btrfs or dir backed container rename destroys rootfs

### ISSUE THREAD

**Issue by Artanicus** [lxc/lxc#2711]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 18.04.1 LTS
 * The output of
   * `lxc-start --version` 3.0.2
   * `lxc-checkconfig`
```
Kernel configuration not found at /proc/config.gz; searching...
Kernel configuration found at /boot/config-4.15.0-38-generic
--- Namespaces ---
Namespaces: enabled
Utsname namespace: enabled
Ipc namespace: enabled
Pid namespace: enabled
User namespace: enabled
Network namespace: enabled

--- Control groups ---
Cgroups: enabled

Cgroup v1 mount points:
/sys/fs/cgroup/systemd
/sys/fs/cgroup/cpu,cpuacct
/sys/fs/cgroup/freezer
/sys/fs/cgroup/pids
/sys/fs/cgroup/hugetlb
/sys/fs/cgroup/cpuset
/sys/fs/cgroup/net_cls,net_prio
/sys/fs/cgroup/perf_event
/sys/fs/cgroup/memory
/sys/fs/cgroup/blkio  
/sys/fs/cgroup/rdma
/sys/fs/cgroup/devices

Cgroup v2 mount points:
/sys/fs/cgroup/unified

Cgroup v1 clone_children flag: enabled
Cgroup device: enabled
Cgroup sched: enabled
Cgroup cpu account: enabled
Cgroup memory controller: enabled
Cgroup cpuset: enabled
--- Misc ---
Veth pair device: enabled, loaded
Macvlan: enabled, not loaded
Vlan: enabled, not loaded
Bridges: enabled, loaded
Advanced netfilter: enabled, not loaded
CONFIG_NF_NAT_IPV4: enabled, loaded
CONFIG_NF_NAT_IPV6: enabled, not loaded
CONFIG_IP_NF_TARGET_MASQUERADE: enabled, loaded
CONFIG_IP6_NF_TARGET_MASQUERADE: enabled, not loaded
CONFIG_NETFILTER_XT_TARGET_CHECKSUM: enabled, loaded
CONFIG_NETFILTER_XT_MATCH_COMMENT: enabled, not loaded
FUSE (for use with lxcfs): enabled, not loaded

--- Checkpoint/Restore ---
checkpoint restore: enabled
CONFIG_FHANDLE: enabled
CONFIG_EVENTFD: enabled
CONFIG_EPOLL: enabled
CONFIG_UNIX_DIAG: enabled
CONFIG_INET_DIAG: enabled
CONFIG_PACKET_DIAG: enabled
CONFIG_NETLINK_DIAG: enabled
File capabilities:

Note : Before booting a new kernel, you can check its configuration
usage : CONFIG=/path/to/config /usr/bin/lxc-checkconfig
```
   * `uname -a` Linux phrenos 4.15.0-38-generic #41-Ubuntu SMP Wed Oct 10 10:59:38 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
   * `cat /proc/self/cgroup`
```
12:devices:/user.slice
11:rdma:/
10:blkio:/user.slice
9:memory:/user.slice
8:perf_event:/
7:net_cls,net_prio:/
6:cpuset:/
5:hugetlb:/
4:pids:/user.slice/user-1000.slice/session-3.scope
3:freezer:/user/root/0
2:cpu,cpuacct:/user.slice
1:name=systemd:/user/root/0
0::/user.slice/user-1000.slice/session-3.scope
```
   * `cat /proc/1/mounts`
```
sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
udev /dev devtmpfs rw,nosuid,relatime,size=32886996k,nr_inodes=8221749,mode=755 0 0
devpts /dev/pts devpts rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 0 0
tmpfs /run tmpfs rw,nosuid,noexec,relatime,size=6583376k,mode=755 0 0
/dev/sdl4 / btrfs rw,noatime,compress=lzo,ssd,space_cache,subvolid=5,subvol=/ 0 0
securityfs /sys/kernel/security securityfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /dev/shm tmpfs rw,nosuid,nodev 0 0
tmpfs /run/lock tmpfs rw,nosuid,nodev,noexec,relatime,size=5120k 0 0
tmpfs /sys/fs/cgroup tmpfs ro,nosuid,nodev,noexec,mode=755 0 0
cgroup /sys/fs/cgroup/unified cgroup2 rw,nosuid,nodev,noexec,relatime 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,nosuid,nodev,noexec,relatime,xattr,name=systemd 0 0
pstore /sys/fs/pstore pstore rw,nosuid,nodev,noexec,relatime 0 0
efivarfs /sys/firmware/efi/efivars efivarfs rw,nosuid,nodev,noexec,relatime 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,nosuid,nodev,noexec,relatime,pids 0 0
cgroup /sys/fs/cgroup/hugetlb cgroup rw,nosuid,nodev,noexec,relatime,hugetlb 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset,clone_children 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup rw,nosuid,nodev,noexec,relatime,net_cls,net_prio 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,nosuid,nodev,noexec,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0 0
cgroup /sys/fs/cgroup/rdma cgroup rw,nosuid,nodev,noexec,relatime,rdma 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs rw,relatime,fd=25,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=26651 0 0
debugfs /sys/kernel/debug debugfs rw,relatime 0 0
mqueue /dev/mqueue mqueue rw,relatime 0 0
hugetlbfs /dev/hugepages hugetlbfs rw,relatime,pagesize=2M 0 0
configfs /sys/kernel/config configfs rw,relatime 0 0
fusectl /sys/fs/fuse/connections fusectl rw,relatime 0 0
/dev/loop0 /snap/core/5328 squashfs ro,nodev,relatime 0 0
/dev/loop1 /snap/core/5548 squashfs ro,nodev,relatime 0 0
/dev/loop2 /snap/core/5662 squashfs ro,nodev,relatime 0 0
/dev/sdl1 /boot ext2 rw,relatime,block_validity,barrier,user_xattr,acl 0 0
/dev/sdl3 /boot/efi vfat rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro 0 0
/dev/sda /ssd btrfs rw,noatime,compress=lzo,ssd,space_cache,subvolid=5,subvol=/ 0 0
/dev/sde1 /mnt/tb btrfs rw,relatime,space_cache,subvolid=5,subvol=/ 0 0
/dev/md0 /srv btrfs rw,relatime,space_cache,subvolid=5,subvol=/ 0 0
lxcfs /var/lib/lxcfs fuse.lxcfs rw,nosuid,nodev,relatime,user_id=0,group_id=0,allow_other 0 0
binfmt_misc /proc/sys/fs/binfmt_misc binfmt_misc rw,relatime 0 0
tmpfs /run/user/1004 tmpfs rw,nosuid,nodev,relatime,size=6583372k,mode=700,uid=1004,gid=1005 0 0
tmpfs /run/user/1000 tmpfs rw,nosuid,nodev,relatime,size=6583372k,mode=700,uid=1000,gid=1000 0 0
/dev/sda /var/lib/lxc btrfs rw,noatime,compress=lzo,ssd,space_cache,subvolid=5,subvol=/lxc 0 0
```

# Issue description

If a container with an old-style implicit btrfs rootfs  (without btrfs: prefix) has a `lxc-copy -n old -R -N new` done on it, an overlay of the rootfs is created and then *the base rootfs is destroyed* making the overlay useless. lxc-update-config does not add the btrfs: -prefix and save a user from this! Adding it manually may allow a rename to be non-destructive but due to #2612 I'm unable to test since lxc-copy is just broken for btrfs backed containers in general.

I also did testing of dir backed containers without a dir: -prefix and the same destruction occurs there. Manually adding the dir: -prefix is a workaround.

# Steps to reproduce

 1. `sudo lxc-create -B btrfs -n base -t download -- -d ubuntu -r bionic -a amd64`
 2. `sed -i -e 's/btrfs://' /var/lib/lxc/base/config`
 3. `lxc-copy -n base -R -N base_maimed -l WARN`
```
lxc-copy: base: storage/btrfs.c: get_btrfs_subvol_path: 100 Failed to append name - rootfs
*snip* repeats many times
```
 4. `ls /var/lib/lxc/base_maimed/rootfs # empty`
 5. `grep rootfs /var/lib/lxc/base_maimed/config`
```
lxc.rootfs.path = overlay:/var/lib/lxc/base/rootfs:/var/lib/lxc/base_maimed/delta0
```

**Comment by brauner** [comment]
The btrfs issue should be fixed but will be in 3.0.3 unless @stgraber uploads a fix to the ubuntu package.

**Comment by brauner** [comment]
> I also did testing of dir backed containers without a dir: -prefix and the same destruction occurs there. Manually adding the dir: -prefix is a workaround.

That works as expected for me no matter if with or without prefix: a simple dir based copy is created and the old container is deleted leaving me with a working dir backed renamed container.
